include/circt/Dialect/ESI/ESIStructure.td

```diff
@@ -39,6 +39,11 @@ def ESIPureModuleOp : ESI_Op<"pure_module",
   let extraClassDeclaration = [{
     // Implement RegionKindInterface.
     static RegionKind getRegionKind(unsigned index) { return RegionKind::Graph;}
+
+    //===------------------------------------------------------------------===//
+    // PortList Methods
+    //===------------------------------------------------------------------===//
+    ::circt::hw::ModulePortInfo getPortList();
```

Comment:
Isn't this redundant? I would imagine this method is already declared via `DeclareOpInterfaceMethods<HWModuleLike>`.

Reply:
It is not.  And if it were, it would be a compile error (duplicate definition).

Reply:
true